src/server/handlers.ts

```diff
@@ -318,44 +306,26 @@ export const addHandlers = (connection: Connection, redirectConsole = true) => {
                             console.log('Loading packages for directory:', dir);
 
                             let overrideMotokoVersion: string | undefined;
-                            try {
-                                const result = await execa(
-                                    'dfx',
-                                    ['--version'],
-                                    {
-                                        cwd: dir,
-                                    },
-                                );
-                                const match = /dfx 0\.(\d+)/.exec(
-                                    result.stdout,
-                                );
-                                if (match) {
-                                    // TODO: generalize to all Motoko versions
-                                    const dfxMinorVersion = +match[1];
-                                    if (dfxMinorVersion < 18) {
-                                        overrideMotokoVersion = '0.10.4';
-                                    }
+                            if (!initializationOptions.useDefaultMocJs) {
+                                const res = await getWorkspaceMocVersion(dir);
+                                if (res.isOk()) {
+                                    overrideMotokoVersion = res.value;
+                                    console.log(
+                                        'Detected Motoko version:',
+                                        overrideMotokoVersion,
+                                    );
+                                } else {
+                                    console.warn(
+                                        'Could not determine Motoko version:',
+                                        res.error.message,
+                                    );
```

Comment:
Perhaps also show where the version was detected. E.g., if I open `test/rename/prepare_rename.mo`, I see: 
```
Loading packages for directory: /home/heitor/Work/dfinity/vscode-motoko/test/definition
Loading packages for directory: /home/heitor/Work/dfinity/vscode-motoko/test/rename
Loading packages for directory: /home/heitor/Work/dfinity/vscode-motoko/test/requestMocJs
Loading packages for directory: /home/heitor/Work/dfinity/vscode-motoko/test/workspace
Detected Motoko version: 0.16.3
Downloading moc-0.16.3.js
Detected Motoko version: 0.13.4
Downloading moc-0.13.4.js
Detected Motoko version: 0.13.4
Downloading moc-0.13.4.js
Detected Motoko version: 0.13.4
Downloading moc-0.13.4.js
Using workspace moc.js version: 0.13.4
```
So it's a bit unclear.